### PR TITLE
WT-9141 Duplicate prefix search tests using the cursor bound logic.

### PR DESCRIPTION
**Pull request by luke-pearson**

### CODE REVIEW

File: test/suite/wtbound.py [hunk 1]
```diff
@@ -28,6 +28,17 @@
 
 import wiredtiger, wttest
 
+def set_prefix_bound(test, cursor, lower_bound):
+    test.assertEqual(cursor.reset(), 0)
+    cursor.set_key(lower_bound)
+    test.assertEqual(cursor.bound("action=set,bound=lower,inclusive=true"), 0)
+    # Strings are immutable in python.
+    upper_bound = list(lower_bound)
+    upper_bound[len(upper_bound) - 1] = chr(ord(upper_bound[len(upper_bound) - 1]) + 1)
```
Comment (jiechenbo) [inline test/suite/wtbound.py:37]: What does chr and ord do here? Could you add a comment explaining as well?

File: test/suite/test_search_near02.py [hunk 2]
```diff
@@ -26,185 +26,68 @@
 # ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
 # OTHER DEALINGS IN THE SOFTWARE.
 #
+# test_search_near02.py
+#       Search_near with a key past the end much like search_near_05, but
```
Comment (jiechenbo) [inline test/suite/test_search_near02.py:30]: np: search_near_05 doesn't exist.

**Review by jiechenbo — APPROVED** [review]
LGTM! Just a few nit pick changes. Do we want to remove the search_near python tests in this ticket or remove it when we deprecate prefix search near?

**Comment by luke-pearson** [comment]
>Do we want to remove the search_near python tests in this ticket or remove it when we deprecate prefix search near?
Some of them aren't related to prefix search near, the one that is will be removed as it doesn't make sense to migrate it.

File: test/suite/wtbound.py (hunk 1 above)
Comment (luke-pearson) [reply test/suite/wtbound.py:37]: There is a comment explaining, string aren't mutable in python so we represent the string as a list, ord converts the char to an int and chr converts it back to a char, we then take the string form of the list and set it as the key.

Comment (luke-pearson) [reply test/suite/wtbound.py:39]: e79451b32

File: test/suite/test_search_near02.py (hunk 2 above)
Comment (luke-pearson) [reply test/suite/test_search_near02.py]: e79451b32

**Review by jiechenbo — APPROVED** [review]
LGTM!

File: test/suite/wtbound.py [hunk 3]
```diff
@@ -28,6 +28,19 @@
 
 import wiredtiger, wttest
 
+def set_prefix_bound(test, cursor, lower_bound):
+    test.assertEqual(cursor.reset(), 0)
+    cursor.set_key(lower_bound)
+    test.assertEqual(cursor.bound("action=set,bound=lower,inclusive=true"), 0)
+    # Strings are immutable in python so we cast the string to a list, then we convert the last
+    # element to an integer using ord() and then convert it back to a char with chr(), finally we
+    # conver the list back to a string.
```
Comment (jiechenbo) [inline test/suite/wtbound.py:37]: np: `convert` instead of conver

**Review by jiechenbo — APPROVED** [review]
LGTM

**Review by clarissecheah — APPROVED** [review]
LGTM.

File: test/suite/test_cursor_bound11.py [hunk 4]
```diff
@@ -0,0 +1,334 @@
...
+        session2.rollback_transaction()
+        session2.begin_transaction('ignore_prepare=true')
+        cursor4 = session2.open_cursor(uri)
+        set_prefix_bound(self, cursor4, 'c')
+        cursor4.set_key('c')
+        self.assertEqual(cursor4.search_near(), 1)
+        bound_skip_count = self.get_stat(stat.conn.cursor_next_skip_total, session2)
+        # We expect to not skip any entries and return 'cc'
+        self.assertEqual(bound_skip_count - skip_count, 0)
+        self.assertEqual(cursor4.get_key(), b'cc')
```
Comment (clarissecheah) [inline test/suite/test_cursor_bound11.py:326]: np: Why not use self.check_key() here?

Comment (luke-pearson) [reply test/suite/test_cursor_bound11.py:326]: 🤔 good point.

Comment (luke-pearson) [reply test/suite/test_cursor_bound11.py:326]: This test wasn't built with `wtbound` functionality in mind, as such check_key doesn't work here.

**Pull request merged**